### DISCUSSION TITLE
Allow state to persist in localStorage by using the zustand persist function

### DISCUSSION THREAD

**Discussion by KevinMusgrave** [xyflow/xyflow#2197] [New Features]
The react-flow state could persist in localStorage by [using the persist function](https://github.com/pmndrs/zustand/wiki/Persisting-the-store's-data). That way the user doesn't have to click any Save or Restore buttons. 

One implementation approach would be to add a boolean `persist` argument and a `persistProps` argument.

**Comment by kahboom** [comment]
This would be awesome, currently looking for this for my project as well.

**Comment by ZakKa89** [comment]
I am trying to implement this feature with zustand myself, would be cool if it works out of the box though!
